tests/convention_test.py

```diff
@@ -0,0 +1,74 @@
+"""Unit tests for the metadata block in project README files."""
+
+from pathlib import Path
+
+import update_projects_table
+
+import metadata_definitions
+import pytest
+
+
+ROOT_PATH = Path(__file__).parent.parent
+
+TEST_CATEGORIES = [
+    (Path(ROOT_PATH / "devops"), "DevOps"),
+    (Path(ROOT_PATH / "reliability"), "Reliability"),
+    (Path(ROOT_PATH / "samples"), "Samples"),
+]
+
+METADATA_VALID = [
+    (metadata_definitions.CATEGORIES, "categories"),
+    (metadata_definitions.INTEGRATIONS, "integrations"),
+]
+
+
+def test_metadata_block_exists():
+    """Checks for required metadata in README files."""
+    for readme_file in sorted(Path(ROOT_PATH).rglob("README.md")):
+        metadata = update_projects_table.extract_metadata(readme_file)
+
+        assert metadata, f"Metadata block is missing in the README file: {readme_file}"
+        assert all(key in metadata for key in metadata_definitions.METADATA), (
+            f"{metadata['title']} metadata missing required fields."
+        )
+        assert metadata["integrations"], (
+            f"{metadata['title']} metadata must have at least one integration."
+        )
+        assert metadata["categories"], (
+            f"{metadata['title']} metadata must have at least one category."
+        )
+
+
+@pytest.mark.parametrize(("allow_list", "metadata_type"), METADATA_VALID)
+def test_metadata_valid(allow_list, metadata_type):
```

Comment:
Don't be afraid to use long descriptive names for test functions - it's a form of self-documentation that can make the docstring unnecessary. For example:
- `test_metadtata_contains_at_least_1_integration`
- `test_metadata_contains_all_fields`
Note that when a test fails Pytest prints the function's name, not the function's docstring.